src/geode/mesh/helpers/convert_surface_mesh.cpp

```diff
@@ -384,20 +414,50 @@ namespace geode
             to_delete[p] = nb_vertices != 3;
             if( nb_vertices > 3 )
             {
-                absl::FixedArray< std::optional< PolygonEdge > > adjacents(
-                    nb_vertices, std::nullopt );
+                using Edge = std::array< index_t, 2 >;
+                absl::flat_hash_map< Edge, PolygonEdge > adjacents;
+                const auto vertices = surface.polygon_vertices( p );
                 for( const auto e : LRange{ nb_vertices } )
                 {
-                    adjacents[e] = surface.polygon_adjacent_edge( { p, e } );
+                    if( const auto adj =
+                            surface.polygon_adjacent_edge( { p, e } ) )
+                    {
+                        adjacents.emplace(
+                            Edge{ vertices[e],
+                                vertices[e + 1 == nb_vertices ? 0 : e + 1] },
+                            adj.value() );
```

Comment:
are there always a value?

Reply:
No but it is protected inside an if block